discovery/discovery.go

```diff
@@ -26,17 +26,15 @@ func Dir() string {
 		return pluginDir
 	}
 
-	usr, err := user.Current()
-	if err != nil {
-		panic(err)
+	home := os.Getenv("HOME")
+	if usr, err := user.Current(); err == nil {
+		home = usr.HomeDir
 	}
-	return path.Join(usr.HomeDir, ".infrakit/plugins")
+	return path.Join(home, ".infrakit/plugins")
 }
 
 // NewPluginDiscovery creates a plugin discovery based on the environment configuration.
-func NewPluginDiscovery() (Plugins, error) {
-	pluginDir := Dir()
```

Comment:
Previously the constructor grabbed the directory from the env `os.Getenv(PluginDirEnvVar)` when it needed to be different. 

I'm not sure I follow what we gain by explicitly passing it to the constructor?
```
os.Setenv("INFRAKIT_PLUGINS_DIR","/my/test/dir")
plugins, err := discovery.NewPluginDiscovery()
```
vs
```
plugins, err := discovery.NewPluginDiscovery("/my/test/dir")
```